src/status_im/mailserver/core.cljs

```diff
@@ -523,17 +533,37 @@
                    :mailserver/pending-requests))})
 
 (fx/defn handle-request-completed
-  [cofx event]
+  [{{:keys [chats] :as db} :db :as cofx}
+   {:keys [requestID lastEnvelopeHash cursor errorMessage] :as event}]
   (when (accounts.db/logged-in? cofx)
-    (let [error (:errorMessage event)]
-      (if (empty? error)
-        (fx/merge
-         cofx
-         {:mailserver/increase-limit []}
-         (update-mailserver-topics {:request-id (:requestID event)
-                                    :cursor     (:cursor event)}))
-
-        (handle-request-error cofx error)))))
+    (if (empty? errorMessage)
+      (let [new-chats       (into []
+                                  (comp (filter :is-active)
+                                        (filter :public?)
+                                        (filter #(= requestID (:join-time-mail-request-id %))))
+                                  (vals chats))
+            chat            (first new-chats)]
+        (if (seq new-chats)
```

Comment:
It's probably clearer if you check for `chat` rather than `new-chats` (although it is true that if `new-chats` are empty, `chat` is also empty).
i.e
`(if chat ..)`

Reply:
got it. yes.

Reply:
done